src/ray/util/event.h

```diff
@@ -122,16 +122,24 @@ class RayEventContext final {
  public:
   static RayEventContext &Instance();
 
+  static RayEventContext &OriginContext();
+
   void SetEventContext(rpc::Event_SourceType source_type,
                        const std::unordered_map<std::string, std::string> &custom_fields =
                            std::unordered_map<std::string, std::string>());
 
-  void SetCustomFields(const std::string &key, const std::string &value);
+  void SetCustomField(const std::string &key, const std::string &value);
 
   void SetCustomFields(const std::unordered_map<std::string, std::string> &custom_fields);
 
+  // Only for test, isn't thread-safe with SetEventContext.
```

Comment:
can we make this private and only use FRIEND_TEST to access this?

Reply:
Good for me!